src/DynamoCore/Models/DynamoModel.cs

```diff
@@ -3531,6 +3519,177 @@ private bool NotesAlreadyLoaded(IEnumerable<ExtraAnnotationViewInfo> notes)
         }
         #endregion
 
+        #region Insert Private Methods
+
+        /// <summary>
+        /// Update inserted workspace elements GUIDs 
+        /// </summary>
+        /// <param name="viewInfo">viewInfo</param>
+        /// <param name="ws">workspace</param>
+        private void UpdateWorkspaceGUIDs(ref ExtraWorkspaceViewInfo viewInfo, ref WorkspaceModel ws)
```

Comment:
I see no need for the `ref` keyword, you are passing classes, and changes you make will  be reflected on the classes - why do you think you need to use `ref`?

Reply:
I did not know that's how passing classes worked, Michael! I feel ashamed but also thankful for this comment, I will remember that.

All of this code is now removed in favor of a much simpler method that operates directly on the json string before deserialization, as you prompted.